### PR TITLE
[17104] Define new versioning and branching model

### PR DESCRIPTION
**Pull request by EduPonz**
This PR defines a new branch model for the repository so that tracking changes in different minor versions of the different suites is easier.

* [x] Define new branch model in README
* [x] Upgrade workflow to cope with new branch model
* [x] Remove warnings coming from Github actions deprecations

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+Each of these suites is distributed by eProsima in the form of Docker images which can be downloaded [here](https://eprosima.com/index.php/downloads-all).
+
+* [Repository's branching model](#repositorys-branching-model)
+* [Versioning and tagging](#versioning-and-tagging)
+
+## Repository's branching model
+
+The repository tracks the version progression of the products contained in each of the suites through three principal branches, one for each suite.
+Furthermore, since each suite is versioned following a SEMVER pattern, there exist tracking branches for each minor version of each suite (for more information of the suites' versioning schemas, please refer to [Versioning and tagging](#versioning-and-tagging)).
+All these branches are captures in the following table:
```
Comment (jsan-rt) [inline README.md]:
```suggestion
All these branches are captured in the following table:
```

**Review by jsan-rt — CHANGES_REQUESTED** [review]
Just a small typo

File: README.md (hunk 1 above)
Comment (EduPonz) [reply README.md]: Done in ffb48d8

**Review by jsan-rt — APPROVED** [review]
LGTM.

**Pull request merged**